src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1986,6 +1987,10 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
     {
         $loader->load('lock.php');
 
+        if (!class_exists(LockMiddleware::class) || !class_exists(LockFactory::class)) {
```

Comment:
Doesn't this block need to be moved to the `registerMessengerConfiguration()` method as the definition won't exist if the Messenger component is not configured?

Reply:
Indeed, I moved it.